### PR TITLE
Refactor: cosign key types

### PR DESCRIPTION
**Pull request by xopham**
Fixes no issue

## Description

- several different types of keys (ecdsa, rsa) and references (email, kms) are being added to cosign and integrated for connaisseur
- the handling and identification of key type is being refactored to improve extension

## Checklist

- [x] PR is rebased to/aimed at branch `develop`
- [x] PR follows [Contributing Guide](../docs/CONTRIBUTING.md)
- [x] Added tests (if necessary)
- [x] Extended README/Documentation (if necessary)
- [x] Adjusted versions of image and Helm chart in `values.yaml` and `Chart.yaml` (if necessary)

### CODE REVIEW

**Review by peterthomassen — CHANGES_REQUESTED** [review]
Nice! Nits only.

For the record, you *may* be able to do achieve a similar thing using new syntax from https://www.python.org/dev/peps/pep-0636/ (starting Python 3.10). However, it would be not what the new syntax is meant for, which reduces clarity, so I'd advise against it.

File: connaisseur/validators/cosign/cosign_validator.py [hunk 1]
```diff
@@ -161,6 +153,27 @@ def __invoke_cosign(self, image, pubkey):
...
+            key = load_key(pubkey).to_pem()  # raises if invalid
+            return ["--key", "/dev/stdin"], {}, key
+        except ValueError:
+            pass
+
+        # key is KMS reference
+        if re.match(r"^\w{2,20}\:\/\/[\w:\/-]{3,255}$", pubkey):
+            return ["--key", pubkey], {}, b""
+
+        msg = "Public key (reference) '{key}' does not match expected patterns."
```
[the remark this answers is not in the document]
Comment (xopham) [reply connaisseur/validators/cosign/cosign_validator.py:174]: actually this is misleading as key is defined below as being `pubkey`, but I agree this is a circular reference and will fix that

[the remark this answers is not in the document]
Comment (peterthomassen) [reply connaisseur/validators/cosign/cosign_validator.py:174]: My bad, sorry! I thought `msg` was a format string, but it's being interpreted by the exception constructor. Never mind!

**Review by peterthomassen — DISMISSED** [review]
lgtm!

**Review by peterthomassen — APPROVED** [review]

**Pull request merged**